### PR TITLE
Remove deprecated methods in CacheClearer class

### PR DESCRIPTION
**Pull request by PierreRambaud**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Remove deprecated methods in CacheClearer class.
| Type?             | improvement
| Category?         | CO
| BC breaks?        | yes
| Deprecations?     | no
| Fixed ticket?     | Fixes #26293
| How to test?      | Tests are green, because nothing use these methods anymore. Check that the cache clearing runs well.
| Possible impacts? | Please indicate what parts of the software we need to check to make sure everything is alright.

## BC breaks

`PrestaShop\PrestaShop\Adapter\Cache\CacheClearer::clearSmartyCache` is no longer available
`PrestaShop\PrestaShop\Adapter\Cache\CacheClearer::clearMediaCache` is no longer available
`PrestaShop\PrestaShop\Adapter\Cache\CacheClearer::clearSymfonyCache` is no longer available
`PrestaShop\PrestaShop\Adapter\Cache\CacheClearer::clearAllCaches` is no longer available
`prestashop.adapter.cache_clearer` service is no longer available

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/28020)

### CODE REVIEW

**Review by matthieu-rolland — DISMISSED** [review]

**Review by eternoendless — DISMISSED** [review]

**Comment by eternoendless** [comment]
How can this PR be tested?

**Review by matthieu-rolland — APPROVED** [review]

File: src/PrestaShopBundle/Resources/config/services/adapter/services.yml [hunk 1]
```diff
@@ -29,14 +29,6 @@ services:
   prestashop.adapter.check_missing_files:
     class: 'PrestaShop\PrestaShop\Adapter\Requirement\CheckMissingOrUpdatedFiles'
 
-  prestashop.adapter.cache_clearer:
```
Comment (Progi1984) [inline src/PrestaShopBundle/Resources/config/services/adapter/services.yml:32]: Please define in your BC that the service has been removed.

File: src/Adapter/File/HtaccessFileGenerator.php [hunk 2]
```diff
@@ -52,11 +52,11 @@ class HtaccessFileGenerator
     /**
      * HtaccessFileGenerator constructor.
      *
-     * @param CacheClearer $cacheClearer
+     * @param CacheClearerInterface $cacheClearer
      * @param Tools $tools
      * @param bool $multipleViewsConfiguration
      */
-    public function __construct(CacheClearer $cacheClearer, Tools $tools, $multipleViewsConfiguration)
+    public function __construct(CacheClearerInterface $cacheClearer, Tools $tools, $multipleViewsConfiguration)
```
Comment (Progi1984) [inline src/Adapter/File/HtaccessFileGenerator.php:59]: Please define in your BC the change of the signature

**Review by Progi1984 — COMMENTED** [review]
Small changes about BC

File: src/Adapter/File/HtaccessFileGenerator.php (hunk 2 above)
Comment (PierreRambaud) [reply src/Adapter/File/HtaccessFileGenerator.php:59]: It's not a BC breaks as the interface still allow the usage of CacheClearer :)

**Review by Progi1984 — APPROVED** [review]

**Review by AureRita — APPROVED** [review]
hi @PierreRambaud 

I tested it, check with and without the pr if there is some differents and launch the automated tests, there is no issue i think

Thank you for your PR

**Comment by kpodemski** [comment]
Thanks @PierreRambaud @AureRita

**Pull request merged**